### ISSUE TITLE
[Support]: Missing objects

### ISSUE THREAD

**Issue by markgoodyear** [blakeblackshear/frigate#3874]
### Describe the problem you are having

Sometimes Frigate misses a car coming on to my drive and has happened both day and night. Here is a screenshot to help show how a car would enter the drive: https://d.pr/i/mnwAJ1. It only happens with the light blue car, so one thing I can think of is the light blue car is able to drive in quicker due to the angle, so perhaps it isn't getting picked up on the drive entance zone (closest to the road). Another is maybe the car tracking is lost due to the white van and then not being picked back up after. Or perhaps a mix of both. Does this seem a possibility, or perhaps something else at play?

### Version

0.11.0-c461c9e (RC 2)

### Frigate config file

```yaml
mqtt:
  host:
  user:
  password:

cameras:
  front:
    ffmpeg:
      hwaccel_args: -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p
      inputs:
        - path: rtsp://...
          roles:
            - detect
            - rtmp
        - path: rtsp://...
          roles:
            - record
    detect:
      width: 704
      height: 480
      fps: 5
    motion:
      threshold: 20 # default 25
      # improve_contrast: True
      mask:
        - 0,174,0,0,74,0,142,0,149,78,181,76,231,84,138,144,0,229
        - 704,159,704,480,558,480,583,133
        - 705,35,705,0,510,0,510,35
    objects:
      track:
        - person
        - car
      filters:
        person:
          min_score: 0.65
        car:
          min_score: 0.65
    zones:
      drive_entrance:
        coordinates: 598,116,601,154,445,124,467,96
        objects:
          - car
      drive:
        coordinates: 573,480,187,480,102,303,86,261,283,118,357,119,445,128,431,150,590,188
        objects:
          - person
    record:
      enabled: True 
      events:
        required_zones:
          - drive_entrance
          - drive
        retain:
          default: 10
          mode: active_objects
          objects:
            car: 2

    snapshots:
      enabled: True
      bounding_box: True
      required_zones:
        - drive_entrance
        - drive
    mqtt:
      required_zones:
        - drive_entrance
        - drive

detectors:
  coral:
    type: edgetpu
    device: usb
```

### Relevant log output

```shell
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-09-17 20:48:16] frigate.app                    INFO    : Starting Frigate (0.11.0-c461c9e)
Starting migrations
[2022-09-17 20:48:16] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-09-17 20:48:16] peewee_migrate                 INFO    : There is nothing to migrate
[2022-09-17 20:48:16] ws4py                          INFO    : Using epoll
[2022-09-17 20:48:16] frigate.app                    INFO    : Output process started: 218
[2022-09-17 20:48:16] frigate.app                    INFO    : Camera processor started for front: 224
[2022-09-17 20:48:16] frigate.app                    INFO    : Capture process started for front: 227
[2022-09-17 20:48:17] ws4py                          INFO    : Using epoll
[2022-09-17 20:48:16] detector.coral                 INFO    : Starting detection process: 216
[2022-09-17 20:48:16] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-09-17 20:48:19] frigate.edgetpu                INFO    : TPU found
[2022-09-17 20:48:29] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43360]
[2022-09-17 20:48:31] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43360]
[2022-09-17 20:51:13] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43362]
[2022-09-17 21:03:25] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43362]
[2022-09-18 08:43:28] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-09-18 08:43:28] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-09-18 08:43:58] root                           INFO    : Detection process didnt exit. Force killing...
[2022-09-18 08:43:58] detector.coral                 INFO    : Starting detection process: 13290
[2022-09-18 08:43:58] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-09-18 08:44:01] frigate.edgetpu                INFO    : TPU found
[2022-09-18 09:22:56] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43364]
[2022-09-18 09:23:39] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43364]
[2022-09-18 12:49:43] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43368]
[2022-09-18 12:49:53] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43368]
[2022-09-18 17:59:49] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-09-18 17:59:49] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-09-18 18:00:19] root                           INFO    : Detection process didnt exit. Force killing...
[2022-09-18 18:00:19] detector.coral                 INFO    : Starting detection process: 23412
[2022-09-18 18:00:19] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-09-18 18:00:22] frigate.edgetpu                INFO    : TPU found
[2022-09-18 19:33:35] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43370]
[2022-09-18 19:40:16] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43370]
[2022-09-19 11:57:41] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-09-19 11:57:41] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-09-19 11:58:11] root                           INFO    : Detection process didnt exit. Force killing...
[2022-09-19 11:58:11] detector.coral                 INFO    : Starting detection process: 42914
[2022-09-19 11:58:11] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-09-19 11:58:14] frigate.edgetpu                INFO    : TPU found
[2022-09-19 14:51:26] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43372]
[2022-09-19 14:51:28] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:41870]
[2022-09-19 14:51:29] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:41870]
[2022-09-19 14:52:49] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43374]
[2022-09-19 14:52:52] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43374]
[2022-09-19 14:55:31] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43372]
[2022-09-19 20:45:26] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43376]
[2022-09-19 20:45:35] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:41872]
[2022-09-19 20:45:52] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:41872]
[2022-09-19 20:46:01] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43376]
[2022-09-19 20:49:20] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43378]
[2022-09-19 20:49:36] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43378]
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[2022-09-19 20:49:36] frigate.video                  ERROR   : front: Unable to read frames from ffmpeg process.
[2022-09-19 20:49:36] frigate.video                  ERROR   : front: ffmpeg process is not running. exiting capture thread...
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-09-19 20:49:43] frigate.app                    INFO    : Starting Frigate (0.11.0-c461c9e)
Starting migrations
[2022-09-19 20:49:43] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-09-19 20:49:43] peewee_migrate                 INFO    : There is nothing to migrate
[2022-09-19 20:49:43] frigate.app                    INFO    : Output process started: 217
[2022-09-19 20:49:43] frigate.app                    INFO    : Camera processor started for front: 220
[2022-09-19 20:49:43] frigate.app                    INFO    : Capture process started for front: 222
[2022-09-19 20:49:43] ws4py                          INFO    : Using epoll
[2022-09-19 20:49:43] ws4py                          INFO    : Using epoll
[2022-09-19 20:49:43] detector.coral                 INFO    : Starting detection process: 215
[2022-09-19 20:49:43] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-09-19 20:49:46] frigate.edgetpu                INFO    : TPU found
[2022-09-19 20:49:53] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43380]
[2022-09-19 20:49:57] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43380]
[2022-09-19 21:07:32] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43382]
[2022-09-19 21:07:35] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43382]
[2022-09-19 21:52:30] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43384]
[2022-09-19 21:52:36] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43384]
[2022-09-19 21:52:36] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43386]
[2022-09-19 21:52:37] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43386]
[2022-09-19 21:52:37] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43388]
[2022-09-19 21:52:37] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43388]
[2022-09-19 21:52:37] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43390]
[2022-09-19 21:52:37] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43390]
[2022-09-19 21:52:37] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43392]
```

### FFprobe output from your camera

```shell
Tried in the Frigate container and on my Mac, couldn't seem to get this to work.
```

### Frigate stats

```json
{
  "detection_fps": 0,
  "detectors": {
    "coral": {
      "detection_start": 0,
      "inference_speed": 9.36,
      "pid": 215
    }
  },
  "front": {
    "camera_fps": 5.1,
    "capture_pid": 222,
    "detection_fps": 0,
    "pid": 220,
    "process_fps": 5.1,
    "skipped_fps": 0
  },
  "service": {
    "latest_version": "0.10.1",
    "storage": {
      "/dev/shm": {
        "free": 133.1,
        "mount_type": "tmpfs",
        "total": 134.2,
        "used": 1.1
      },
      "/media/frigate/clips": {
        "free": 83890.4,
        "mount_type": "ext4",
        "total": 102221.2,
        "used": 13847.4
      },
      "/media/frigate/recordings": {
        "free": 83890.4,
        "mount_type": "ext4",
        "total": 102221.2,
        "used": 13847.4
      },
      "/tmp/cache": {
        "free": 991.1,
        "mount_type": "tmpfs",
        "total": 1000,
        "used": 8.9
      }
    },
    "temperatures": {},
    "uptime": 1072,
    "version": "0.11.0-c461c9e"
  }
}
```

### Operating system

Proxmox

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Amcrest IP8M-T2599EW

### Any other information that may be helpful

Proxmox setup:

VM with HAOS on + Frigate proxy addon.
LXC with Debian + Docker, this is where Frigate runs.

**Comment by NickM-27** [comment]
For better motion detection especially at night `improve_contrast` is recommended to be enabled. 

Your camera is currently running 5 fps (the recommended value) but that really isn't a lot with a car coming out from behind the van. I'd recommend trying to increase the fps to 8 and see if that makes a difference.

**Comment by markgoodyear** [comment]
@NickM-27 Thanks, I'll try bump the frame rate see if it helps. I did have `improve_contrast enabled`, however that didn't help unfortunately, plus it happens in the day time too (happened today when it was daylight). I'll monitor over the next few days see if it happens again with the frame rate at 8 and report back, it'd usually happen at least once in a few days.

**Comment by markgoodyear** [comment]
Same happened at 8 FPS too, also missed two people getting out of the car too. I've adjusted the zones a little so the drive entrance zone is a bit bigger (https://d.pr/i/H5gDbB), and re-enabled `improve_contrast` as well, just incase it helps with the missing people.

**Comment by NickM-27** [comment]
What do your motion masks look like? At this point it is making me think the motion masks are causing this issue

**Comment by markgoodyear** [comment]
Here's the masks I have: https://d.pr/i/asGNXt.

The neighbour on the left hand side can be detected in the main drive zone due to the motion box size, and also masks a tree, could certainly be reduced. Right hand side similar thing. Anything that looks like it should be adjusted?

I wonder on the tracking of cars on the road too, the motion boxes can go off the top of the screen and looks like it loses cars towards the middle top (I'll try grab a video of that tomorrow when there's traffic), perhaps that could have an affect, it'll be easy to adjust the camera angle if so.

**Comment by NickM-27** [comment]
That does look alright as far as the mask goes, at this point the easiest way to tell would probably be to watch the debug view within frigate with object and motion boxes on and see what goes on.

**Comment by blakeblackshear** [comment]
You could also try making the entrance zone larger by extending further into the driveway.

**Comment by markgoodyear** [comment]
@blakeblackshear Sure, I can extend even further, zones are now: https://d.pr/i/ZWYYZH. Also moved the time down so there's one less motion mask, shouldn't make difference in this scenario, but makes sense to move it to where a mask already is. 

The car will be arrving in the dark later tonight so I'll try monitor it coming in.

**Comment by markgoodyear** [comment]
I managed to watch the car come in undetected until it had stopped on the drive, frustratingly I missed getting a screen recording of it, I'll try get next time.

**Comment by NickM-27** [comment]
@markgoodyear If you are going to do that, make sure that motion boxes are enabled too

**Comment by blakeblackshear** [comment]
It could also be that the object isn't a confirmed true positive before it crosses. It takes at least 3 frames where the median score is above the threshold.

**Comment by markgoodyear** [comment]
Managed to capture it tonight: https://d.pr/v/IrZK9z

I'd guess that the headlights are causing the issue here and I'm not sure if there's much that can be done about it without moving the camera. It has done it in daylight though, however I've been unable to test it with the larger entrance zone since it gets dark here early now. I wonder if turning off the IR on the camera could work better.

There's always the option of unrestricting the `drive` zone, but will that make more recordings and events if the car is lost and re-detected?

**Comment by NickM-27** [comment]
Seems like it has enough frames to get detected so I'm not sure, seems odd but IR is difficult. During the day would be more interesting if it happened because to me that would present evidence of a different issue. You could maybe try lowering the threshold of car and see if that helps.

**Comment by markgoodyear** [comment]
I've dialed down the intensity of the IR right down so the whites are less blown out incase it helps for the night detection. I just spotted the car wasn't detected whilst parked so I've lowered the threshold to 55%. It's picked back up now settled at 60% so that could do the trick also. I should be able to monitor some day time activity tomrrow and over the weekend. Appreciate the help so far trying to figure this out, determined to get to the bottom of it!
